pytorch_blade/pytorch_blade/quantization/pybind_functions.cpp

```diff
@@ -57,16 +62,153 @@ void remove_placeholder(Module& model) {
   }
 }
 
+void replace_aten_fake_quant_with_custom_version(Module& model) {
+  auto g = model.get_method("forward").graph();
+  // the graph should be inlined first
+  Inline(*g);
+  Symbol sym = Symbol::fromQualString(
+      torch::blade::quantization::custom_fake_quant_name);
+  std::vector<Node*> aten_fake_quant_node;
+
+  for (auto n : g->nodes()) {
+    std::string node_kind_str = n->kind().toQualString();
+
+    if (node_kind_str ==
+            std::string(torch::blade::quantization::
+                            at_fake_quant_per_channel_affine_name) ||
+        node_kind_str ==
+            std::string(torch::blade::quantization::
+                            at_fake_quant_per_tensor_affine_name)) {
+      // aten::fake_quant will be destroyed later, collect it here
+      aten_fake_quant_node.push_back(n);
+
+      auto input_type = n->inputs()[0]->type()->cast<c10::TensorType>();
+      TORCH_CHECK(input_type, "Unexpected input type of aten::fake_quant.")
+      auto device = input_type->device();
+
+      // create torch_blade_fake_quant custom op
+      Node* fake_quant_node = g->insertNode(g->create(sym));
+      fake_quant_node->moveAfter(n);
+      fake_quant_node->output()->setType(n->inputs()[0]->type());
+      n->output()->replaceAllUsesWith(fake_quant_node->output());
+
+      Value* inputs = n->inputs()[0]; // inputs
+
+      // scale & zero_zero point could be scalar in
+      // aten::fake_quantize_per_tensor_affine. Convert it to tensor to match
+      // our schema.
+      Value* scales = n->inputs()[1];
+      if (!scales->type()->isSubtypeOf(at::TensorType::get())) {
+        TORCH_CHECK(
+            scales->type()->isSubtypeOf(c10::NumberType::get()),
+            "Unexpected scalar type for scales.");
+        float scale_num = scales->node()->f(attr::value);
+        auto tensor_option =
+            torch::TensorOptions().dtype(torch::kFloat).device(device);
+        Tensor scale_tensor = torch::tensor(scale_num, tensor_option);
+        scales = insert_prim_constant<torch::Tensor>(
```

Comment:
template type for calling insert_prim_constant can be omitted?

Reply:
done